tsdb/chunks/head_chunks.go

```diff
@@ -376,7 +376,29 @@ func repairLastChunkFile(files map[int]string) (_ map[int]string, returnErr erro
 	if err != nil {
 		return files, errors.Wrap(err, "file stat during last head chunk file repair")
 	}
-	if info.Size() == 0 {
+
+	remove := info.Size() == 0 // Empty file.
+	if info.Size() >= MagicChunksSize {
+		// Check if magic number is 0.
+		f, err := os.Open(files[lastFile])
+		if err != nil {
+			return files, errors.Wrap(err, "open file during last head chunk file repair")
+		}
+
+		buf := make([]byte, MagicChunksSize)
+		_, err = f.Read(buf)
+		if err != nil {
+			return files, errors.Wrap(err, "failed to read magic number during last head chunk file repair")
+		}
+
+		remove = binary.BigEndian.Uint32(buf) == 0
```

Comment:
Should this remove anything that is not `MagicHeadChunks`? I would imagine an incorrect magic number would also be a sign of corruption, not just 0.

Reply:
If the magic number was fully present and not equals then it is caught at some other place that checks for file corruptions, so it should be already taken care of.

Reply:
Looks like the snapshot replay relies on the genuine corruption being reported up the function call so that we can discard the snapshot. So I think we should not take care of different magic number here. However, this gave me an idea that we are missing another case: if the number of bytes in the file was 1-3, then we say it is not corrupted while it should be deleted here (because size is not 0 and we dont check magic number in that case). I will fix that.

Reply:
Good to know about the genuine corruption being handled elsewhere, thanks for adding the comment.